### PR TITLE
fix type errors for alerting components with new react-hook-form version

### PR DESCRIPTION
**Pull request by nathanrodman**
**What this PR does / why we need it**:

Fixes type errors for react-hook-form in the ReceiverForm and AlertTypeStep components, mostly through type casting because react-hook-form can get messy with controlled inputs that have an object type.

**Which issue(s) this PR fixes**:

Fixes issues in #43670

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by gabor** [comment]
( don't you want the milestone be 8.5.0? the related package-update will only happen on main-branch so i don't think there's a need for backporting, but i might be missing something)

**Comment by gillesdemey** [comment]
That's correct @gabor, thanks for catching it!

**Review by gillesdemey — APPROVED** [review]
LGTM; only one non-blocking nit – I haven't found a better way to get around the `as R` typecasting but we don't have to solve that in this PR ✌️

File: public/app/features/alerting/unified/components/receivers/form/ReceiverForm.tsx [hunk 1]
```diff
@@ -94,7 +94,7 @@ export function ReceiverForm<R extends ChannelValues>({
           Because there is no default policy configured yet, this contact point will automatically be set as default.
         </Alert>
       )}
-      <form onSubmit={handleSubmit(submitCallback, onInvalid)}>
+      <form onSubmit={handleSubmit(submitCallback as SubmitHandler<ReceiverFormValues<R>>, onInvalid)}>
```
[the remark this answers is not in the document]
Comment (nathanrodman) [reply public/app/features/alerting/unified/components/receivers/form/ReceiverForm.tsx:97]: I've tried that and get some type incompatibility on the `items` passed into `onSubmit` within `handleSubmit`

```
Types of property 'items' are incompatible.
        Type 'UnpackNestedValue<ControlledField<R>>[]' is not assignable to type 'ControlledField<R>[]'.
```

/shrug

**Review by konrad147 — APPROVED** [review]

**Comment by gillesdemey** [comment]
LGTM!

**Comment by gabor** [comment]
hi, can this be merged?

**Comment by gillesdemey** [comment]
This is one good to go 👍

**Pull request merged**